File: examples/python/core/callbacks/multiple-callbacks.py
@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Wait for multiple external systems."""
    # Create callbacks for different systems
    credit_check = context.create_callback(
        name="credit_check",
        config=CallbackConfig(timeout=Duration.from_minutes(5)),
    )
    
    fraud_check = context.create_callback(
        name="fraud_check",
        config=CallbackConfig(timeout=Duration.from_minutes(3)),
    )
    
    # Send requests to external systems
    request_credit_check(credit_check.callback_id, event["customer_id"])
    request_fraud_check(fraud_check.callback_id, event["transaction_data"])
    
    # Wait for both results
    credit_result = credit_check.result()
    fraud_result = fraud_check.result()
    
    # Make decision based on both checks
    approved = (
        credit_result.get("score", 0) > 650 and
        fraud_result.get("risk_level") == "low"
    )
    
    return {
        "approved": approved,
        "credit_score": credit_result.get("score"),
        "fraud_risk": fraud_result.get("risk_level"),
    }

File: examples/java/core/callbacks/multiple-callbacks.java
// Create callbacks for different systems
var creditCheck = ctx.createCallback("credit_check", Map.class,
    CallbackConfig.builder()
        .timeout(Duration.ofMinutes(5))
        .build());

var fraudCheck = ctx.createCallback("fraud_check", Map.class,
    CallbackConfig.builder()
        .timeout(Duration.ofMinutes(3))
        .build());

// Send requests to external systems
ctx.step("request_checks", Void.class, stepCtx -> {
    requestCreditCheck(creditCheck.callbackId(), event.get("customer_id"));
    requestFraudCheck(fraudCheck.callbackId(), event.get("transaction_data"));
    return null;
});

// Wait for both results
var creditResult = creditCheck.get();
var fraudResult = fraudCheck.get();

// Make decision based on both checks
var approved = ((Number) creditResult.get("score")).intValue() > 650
    && "low".equals(fraudResult.get("risk_level"));

return Map.of(
    "approved", approved,
    "credit_score", creditResult.get("score"),
    "fraud_risk", fraudResult.get("risk_level"));
